plugins/modules/secrets_provider_vault.py

```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2020, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["stableinterface"],
+    "supported_by": "certified",
+}
+
+DOCUMENTATION = '''
+module: secrets_provider_vault
+author:
+  - Aljaz Kosir (@aljazkosir)
+  - Manca Bizjak (@mancabizjak)
+  - Miha Dolinar (@mdolin)
+  - Tadej Borovsak (@tadeboro)
+short_description: Manage Sensu VaultProvider secrets providers
+description:
+  - Create, update or delete a Sensu Go VaultProvider secrets provider.
+  - For more information, refer to the Sensu Go documentation at
+    U(https://docs.sensu.io/sensu-go/latest/operations/manage-secrets/secrets-providers/).
+version_added: 1.6.0
+extends_documentation_fragment:
+  - sensu.sensu_go.auth
+  - sensu.sensu_go.name
+  - sensu.sensu_go.state
+options:
+  address:
+    description:
+      - Address of the Vault server.
+      - Required if I(state) is C(present).
+    type: str
+  token:
+    description:
+       - Authentication token to use with Vault.
+       - Required if I(state) is C(present).
+    type: str
+  version:
+    description:
+      - Version of the Vault key/value store.
+      - Please refer to U(https://www.vaultproject.io/docs/secrets/kv) for
+          additional information.
+      - Required if I(state) is C(present).
+    type: str
+    choices: [v1, v2]
+  timeout:
+    description:
+      - Timeout (in seconds) for connection to Vault server.
+    type: int
+  max_retries:
+    description:
+      - Maximum number of times to retry failed connections to Vault server.
+    type: int
+  rate_limit:
+    description:
+      - Maximum number of secrets requests for per second.
+    type: float
+  burst_limit:
+    description:
+      - Maximum allowed number of secrets requests in a rate interval.
+    type: int
+  tls:
+    description:
+      - TLS configuration for establishing connection with Vault server.
+    type: dict
+    suboptions:
+      ca_cert_file:
+        description:
+          - Path to the certificate file of the trusted certificate authority.
+        type: str
+      client_cert_file:
+        description:
+          - Path to the client certificate file.
+        type: str
+      client_key_file:
+        description:
+          - Path to the client key file.
+        type: str
+      common_name:
+        description:
+          - Common name of the Vault server.
```

Comment:
I would assume this is common name of the client. And I say assume because I am not sure even after reading the Sensu and Vault docs (so much about documentation intuitiveness). But the server's common name has no use here, so I guess this is client-related value.